### PR TITLE
DEP: remove deprecated `alen` and `asscalar` functions

### PR DESCRIPTION
**Pull request by rgommers**
`alen` has been removed since 1.18.0, and `asscalar` since 1.16.0

### CODE REVIEW

**Comment by WarrenWeckesser** [comment]
One more occurrence of `alen` to remove:

https://github.com/numpy/numpy/blob/d1e711fcb0d8132d607698889db6de6c3adedd8d/tools/functions_missing_types.py#L35

**Comment by rgommers** [comment]
> One more occurrence of alen to remove:

Good catch, thanks. Fixed and added a release note.

File: doc/release/upcoming_changes/20414.expired.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+``alen`` and ``asscalar`` removed
+---------------------------------
+
+The derecated ``np.alen`` and ``np.asscalar`` functions were removed.
```
Comment (WarrenWeckesser) [inline doc/release/upcoming_changes/20414.expired.rst:4]: Typo:
```suggestion
The deprecated ``np.alen`` and ``np.asscalar`` functions were removed.
```

Comment (rgommers) [reply doc/release/upcoming_changes/20414.expired.rst:4]: thanks, good catch - fixed. not rerunning CI for a one-letter typo fix.

**Review by WarrenWeckesser — APPROVED** [review]
Thanks @rgommers.

**Pull request merged**